src/core/reference/include/openvino/reference/scaled_dot_product_attention.hpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include <cmath>
+
+#include "openvino/core/shape.hpp"
+#include "openvino/reference/add.hpp"
+#include "openvino/reference/matmul.hpp"
+#include "openvino/reference/multiply.hpp"
+#include "openvino/reference/softmax.hpp"
+
+namespace helpers {
+template <typename T>
+std::vector<T> CreateCausalAttentionMask(const ov::Shape& shape) {
+    std::vector<T> maskData(shape_size(shape), std::numeric_limits<T>::lowest());
+    auto L = shape[shape.size() - 2];
+    auto S = shape[shape.size() - 1];
+    for (size_t i = 0; i < L; ++i) {
+        size_t j = 0;
+        while (i >= j && j < S) {
+            maskData[i * S + j] = 0;
+            ++j;
+        }
+    }
+    return maskData;
+}
+
+template <typename T>
+std::vector<T> CreateAttentionMaskFromOvBoolean(const char* maskBool, const ov::Shape& shape) {
+    std::vector<T> maskData(shape_size(shape));
+
+    std::transform(maskBool, maskBool + shape_size(shape), maskData.begin(), [](char val) {
+        if (val == 0) {
+            return std::numeric_limits<T>::lowest();
+        } else {
+            return static_cast<T>(0);
+        }
+    });
+
+    return maskData;
+}
+}  // namespace helpers
+namespace ov {
+namespace reference {
+template <typename T, typename TMask>
+void scaled_dot_product_attention(const T* query,
+                                  const T* key,
+                                  const T* value,
+                                  const TMask* mask,
+                                  const T* scale,
+                                  T* output,
+                                  bool isCausal,
+                                  const Shape& queryShape,
+                                  const Shape& keyShape,
+                                  const Shape& valueShape,
+                                  const Shape& maskShape,
+                                  const Shape& outputShape) {
```

Comment:
```suggestion
                                  bool is_causal,
                                  const Shape& query_shape,
                                  const Shape& key_shape,
                                  const Shape& value_shape,
                                  const Shape& mask_shape,
                                  const Shape& output_shape) {
```
Detail: Use `snake_case` for variables